common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -70,6 +70,11 @@
 
   private static final String RECOVERY_FILE_NAME = "registeredExecutor.ldb";
 
+  // Whether failure during service initialization should stop the NM.
+  @VisibleForTesting
+  static final String STOP_ON_FAILURE_KEY = "spark.yarn.shuffle.stopOnFailure";
```

Comment:
it would be nice to document this so others now about it.   There is a section in job scheduling doc about the external shuffle service, otherwise maybe just put in the yarn section.  We probably should have pointer from yarn section to the job scheduler section or just a standalone section on the shuffle service, but that is a different issue.